src/browser/PasskeyUtils.cpp

```diff
@@ -352,15 +352,11 @@ ExtensionResult PasskeyUtils::buildExtensionData(QJsonObject& extensionObject) c
     return {};
 }
 
-QJsonObject PasskeyUtils::buildClientDataJson(const QJsonObject& publicKey, const QString& origin, bool get) const
+// Serialization order: https://w3c.github.io/webauthn/#clientdatajson-serialization
+QString PasskeyUtils::buildClientDataJson(const QJsonObject& publicKey, const QString& origin, bool get) const
 {
-    QJsonObject clientData;
-    clientData["challenge"] = publicKey["challenge"];
-    clientData["crossOrigin"] = false;
-    clientData["origin"] = origin;
-    clientData["type"] = get ? QString("webauthn.get") : QString("webauthn.create");
-
-    return clientData;
+    return QString("{\"type\":\"%1\",\"challenge\":\"%2\",\"origin\":\"%3\",\"crossOrigin\":false}")
+        .arg((get ? QString("webauthn.get") : QString("webauthn.create")), publicKey["challenge"].toString(), origin);
```

Comment:
Since the values that can go in here are very limited, this is probably fine, but it feels icky to do it this way. Couldn't we implement a custom serialisation implementation similar to the ECMAScript example in the JCS spec? https://www.rfc-editor.org/rfc/rfc8785#name-ecmascript-sample-canonical

Reply:
I think this is good enough for what we want to do. If we need to do this anywhere else than we can introduce a serializer.

Reply:
I suppose challenge is base64, so it won't ever contain any invalid characters, right?

Reply:
Origin is a url like `https://webauthn.io`